### PR TITLE
[IMP] web, project: add accordion menu component

### PR DESCRIPTION
**Pull request by rfr-odoo**
On the small screen, the sub dropdown is a bit weird to positioning, so we use an accordion instead.
We apply this change also on desktop to avoid split behavior between small screen and bigger screen and also to avoid duplicating code.

Note: some tests were adapted to reflect the change

Task-ID: 3276542 (Part of)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/dropdown/dropdown_navigation_hook.js [hunk 1]
```diff
@@ -70,7 +75,9 @@ export function useDropdownNavigation() {
     const menuRef = useRef("menuRef");
     /** @type {MenuElement[]} */
     let menuElements = [];
-    useEffect(() => {
+
+    let cleanupDropdownNavigation;
+    const setupDropdownNavigation = () => {
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/dropdown_navigation_hook.js:80]: Please rename to something like:
```suggestion
    let cleanupMenuElements;
    const refreshMenuElements = () => {
```

File: addons/web/static/src/core/dropdown/dropdown_navigation_hook.js [hunk 2]
```diff
@@ -168,6 +175,19 @@ export function useDropdownNavigation() {
                 }
             }
         };
+        return () => cleanupDropdownNavigation();
+    };
+
+    useEffect(setupDropdownNavigation);
+
+    // Set up nested accordion
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/dropdown_navigation_hook.js:183]: Maybe:
```
// Set up nested accordion
// This is needed in order to keep the parent dropdown
// aware of the accordion menu elements when its state has changed.
```

File: addons/web/static/src/core/dropdown/dropdown_navigation_hook.js
Comment (brboi) [inline addons/web/static/src/core/dropdown/dropdown_navigation_hook.js:1]: I only reviewed this file ;-)

File: addons/web/static/src/core/dropdown/accordion_item.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    <t t-name="web.AccordionItem" owl="1">
+        <div class="o_accordion position-relative">
+            <button class="o_menu_item o_accordion_toggle dropdown-item" tabindex="0"
+                    t-att-class="{'selected': props.selected}"
+                    t-esc="props.description"
+                    t-on-click="() => state.open = !state.open"
+            />
+            <t t-if="state.open">
+                <div class="o_accordion_values ms-4 border-start" t-att-aria-expanded="state.open ? 'true' : 'false'">
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/accordion_item.xml:12]: I suppose you can replace `t-att-aria-expanded="..."` with `aria-expanded="true"`, as this node is inside a `t-if="state.open"`

File: addons/web/static/src/core/dropdown/accordion_item.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    static props = {
+        slots: {
+            type: Object,
+            shape: {
+                default: {},
+            },
+        },
+        description: String,
+        selected: Boolean,
+    };
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/accordion_item.js:29]: Nitpicking about coding style:
not sure if we want to enforce this practice, but generally we place static members of a class above the rest of its members.

File: addons/web/static/src/core/dropdown/accordion_item.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.o_accordion {
+    > button {
```
Comment (brboi) [inline addons/web/static/src/core/dropdown/accordion_item.scss:2]:
```suggestion
.o_accordion_toggle {

**Review by brboi — COMMENTED** [review]
Looks good, I left only nitpicking comments ;-)

**Review by pparidans — APPROVED** [review]
LGTM
@robodoo r+

**Pull request closed without merging**